eZ/Publish/Core/Limitation/ObjectStateLimitationType.php

```diff
@@ -236,8 +236,49 @@ public function getCriterion(APILimitationValue $value, APIUserReference $curren
             return new Criterion\ObjectStateId($value->limitationValues[0]);
         }
 
-        // several limitation values: IN operation
-        return new Criterion\ObjectStateId($value->limitationValues);
+        $groupedLimitationValues = $this->groupLimitationValues($value->limitationValues);
+
+        if (count($groupedLimitationValues) === 1) {
+            // several limitation values: IN operation
+            return new Criterion\ObjectStateId($value->limitationValues);
+        }
+
+        $criterions = [];
+        foreach ($groupedLimitationValues as $limitationGroup) {
+            if (count($limitationGroup) === 1) {
```

Comment:
do we need this, I know we have it other places, but logic wise both sql and solr will be fine with either case right? If so we actually don't need to optimize this here.